src/index.js

```diff
@@ -5,52 +5,68 @@ const SCRIPT_ID = 'hcaptcha-api-script-id';
 const HCAPTCHA_LOAD_FN_NAME = 'hcaptchaOnLoad';
 
 // Prevent loading API script multiple times
-let resolveFn;
-let rejectFn;
-const mountPromise = new Promise((resolve, reject) => {
-  resolveFn = resolve;
-  rejectFn = reject;
-});
+const scripts = [];
 
 // Generate hCaptcha API script
 const mountCaptchaScript = (params={}) => {
   const parent = params.scriptLocation || document.head;
+
   delete params.scriptLocation;
 
   const doc = parent.ownerDocument || document;
+  const win = doc.defaultView || doc.parentWindow;
+
+  const script = scripts.find(({ scope }) => scope === win);
 
-  if (doc.getElementById(SCRIPT_ID)) {
+  if (doc.getElementById(SCRIPT_ID) && script) {
     // API was already requested
-    return mountPromise;
+    return script.promise;
   }
 
-  // Create global onload callback
-  window[HCAPTCHA_LOAD_FN_NAME] = resolveFn;
+  const promise = new Promise((resolve, reject) => {
+    // Create global onload callback
+    win[HCAPTCHA_LOAD_FN_NAME] = resolve;
+
+    const domain = params.apihost || "https://js.hcaptcha.com";
+    delete params.apihost;
 
-  const domain = params.apihost || "https://js.hcaptcha.com";
-  delete params.apihost;
+    const script = doc.createElement("script");
+    script.id = SCRIPT_ID;
+    script.src = `${domain}/1/api.js?render=explicit&onload=${HCAPTCHA_LOAD_FN_NAME}`;
 
-  const script = doc.createElement("script");
-  script.id = SCRIPT_ID;
-  script.src = `${domain}/1/api.js?render=explicit&onload=${HCAPTCHA_LOAD_FN_NAME}`;
+    script.async = params.loadAsync !== undefined? params.loadAsync : true;
+    delete params.loadAsync;
 
-  script.async = params.loadAsync !== undefined? params.loadAsync : true;
-  delete params.loadAsync;
+    script.onerror = (event) => reject('script-error');
 
-  script.onerror = (event) => rejectFn('script-error');
+    const query = generateQuery(params);
+    script.src += query !== ""? `&${query}` : "";
 
-  const query = generateQuery(params);
-  script.src += query !== ""? `&${query}` : "";
+    parent.appendChild(script);
+  });
 
-  parent.appendChild(script);
-  return mountPromise;
+  scripts.push({ promise, scope: win });
+
+  return promise;
 };
 
 
 class HCaptcha extends React.Component {
     constructor (props) {
       super(props);
 
+      const parent = this.props.scriptLocation || document.head;
+      const doc = parent.ownerDocument || document;
+      const win = doc.defaultView || doc.parentWindow;
```

Comment:
how about renaming `win` to `hcaptchaWindow`?

Reply:
I wouldn't consider hCaptcha's window, its still the window maybe frameWindow is better here. This should be addressed here [3f12a29](https://github.com/hCaptcha/react-hcaptcha/pull/185/commits/3f12a29d517a2821b4ef5bad72bd075b544fe5c9)

Reply:
It's not always the frame though
this naming is a bit confusing to me

maybe, `hcaptchaApiWindow` or `apiWindow` would suit better

Reply:
Thats why I do 

```
const frame = getFrame();
// frame.document
// frame.window
```